lib/pure/os.nim

```diff
@@ -2088,6 +2088,216 @@ iterator walkDir*(dir: string; relative=false): tuple[kind: PathComponent, path:
               k = getSymlinkFileKind(path)
             yield (k, y)
 
+type OpenDirStatus* = enum
```

Comment:
convention is to put `type` on its own line

Reply:
when the block contains a single type, it's quite common to write
`type Foo = ...` instead of 
```nim
type
  Foo = ...
```
and is in fact easier to search.
the main advantage of the latter form is for mutually recursive types

Reply:
It's common, but not in Nim's codebase.

Reply:
OK